src/components/QuestionnaireForm/QuestionnaireForm.jsx

```diff
@@ -83,6 +85,19 @@ export default class QuestionnaireForm extends Component {
 
   componentDidMount() {
     this.loadAndMergeForms(this.state.savedResponse);
+
+    document.addEventListener('change', event => {
+      if(event.target.id != "filterCheckbox" && event.target.id != "attestationCheckbox") {
```

Comment:
minor point, the event listener will attempt to filter every time anything is changed, could use one of the props to check to see if the app is currently filtering and avoid the unnecessary calls.

Reply:
Added check the app filter state and validity of the form before filter.